### PR TITLE
Fix: Accessing dead object error in Firefox

### PR DESCRIPTION
**Pull request by MoumitaM**
## Description of the change

> Added error handling for accessing dead object error in Firefox

## Type of change
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [x] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

---
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/567)

### CODE REVIEW

File: utils/utils.js [hunk 1]
```diff
@@ -193,7 +193,12 @@ function getDefaultPageProperties() {
 }
 
 function getReferrer() {
-  return document.referrer || "$direct";
+  // This error handling is in place to avoid accessing dead object(document)
+  try {
+    return document.referrer || "$direct";
+  } catch(e){
+    return "$direct";
```
Comment (saikumarrs) [inline utils/utils.js:200]: Log the error here.

Comment (saikumarrs) [inline utils/utils.js:200]: Put `$direct` into a constant.

File: utils/utils.js [hunk 2]
```diff
@@ -193,7 +193,12 @@ function getDefaultPageProperties() {
 }
 
 function getReferrer() {
-  return document.referrer || "$direct";
+  // This error handling is in place to avoid accessing dead object(document)
+  try {
+    return document.referrer || "$direct";
+  } catch(e){
```
Comment (saikumarrs) [inline utils/utils.js:199]:
```suggestion
  } catch(e) {
```

File: utils/utils.js (hunk 1 above)
Comment (MoumitaM) [reply utils/utils.js]: Addressed

Comment (MoumitaM) [reply utils/utils.js]: Addressed

File: utils/utils.js [hunk 3]
```diff
@@ -193,7 +193,14 @@ function getDefaultPageProperties() {
 }
 
 function getReferrer() {
-  return document.referrer || "$direct";
+  // This error handling is in place to avoid accessing dead object(document)
+  const referrer = "$direct";
```
Comment (saikumarrs) [inline utils/utils.js:197]: A bit more appropriate.
```suggestion
  const defaultReferrer = "$direct";
```

**Review by saikumarrs — APPROVED** [review]

**Pull request merged**